src/sentry/models/dashboard_permissions.py

```diff
@@ -16,17 +29,27 @@ class DashboardPermissions(Model):
     __relocation_scope__ = RelocationScope.Organization
 
     is_creator_only_editable = models.BooleanField(default=False)
+    is_editable_by_everyone = models.BooleanField(default=True)
+    teams_with_edit_access = models.ManyToManyField(
+        "sentry.Team", through=DashboardPermissionsTeam, blank=True
+    )
+
     dashboard = models.OneToOneField(
         "sentry.Dashboard", on_delete=models.CASCADE, related_name="permissions"
     )
 
-    def has_edit_permissions(self, userId):
-        if not self.is_creator_only_editable:
+    def has_edit_permissions(self, user_id):
+        if self.is_editable_by_everyone:
             return True
-        return userId == self.dashboard.created_by_id
+        if user_id == self.dashboard.created_by_id:
+            return True  # Dashboard creator will always have edit perms
+        for team in self.teams_with_edit_access.all():
+            if user_id in team.get_member_user_ids():
+                return True
+        return False
 
     class Meta:
         app_label = "sentry"
         db_table = "sentry_dashboardpermissions"
 
-    __repr__ = sane_repr("is_creator_only_editable")
+    __repr__ = sane_repr("is_editable_by_everyone")
```

Comment:
It would be useful to also add the teams here as well as another arg, so it will print out if we look at it in the shell.

Reply:
Hmm, I think we could add it since the ideal case would be to have only a handful of teams with access. In the case that `is_editable_by_everyone` is true, it's gonna be empty anyway.

Reply:
Ok nvm, looks like it returns the ManyToMany manager. 
<img width="1035" alt="Screenshot 2024-11-04 at 5 10 46 PM" src="https://github.com/user-attachments/assets/ee053c6d-25bf-43e3-b347-30086fbd221e">